ql/ql/src/codeql_ql/style/docs/NonUSSpellingQuery.qll

```diff
@@ -0,0 +1,24 @@
+predicate non_us_word(string wrong, string right) {
+  exists(string s |
+    wrong = s.splitAt("/", 0) and
+    right = s.splitAt("/", 1) and
+    s =
+      [
+        "colour/color", "authorise/authorize", "analyse/analyze", "behaviour/behavior",
+        "modelling/modeling", "modelled/modeled"
+      ]
+  )
+}
+
+bindingset[s]
+predicate contains_non_us_spelling(string s, string wrong, string right) {
+  non_us_word(wrong, right) and
+  (
+    s.matches("%" + wrong + "%") and
+    wrong != "analyse"
+    or
+    // analyses (as a noun) is fine
+    s.regexpMatch(".*analyse[^s].*") and
+    wrong = "analyse"
```

Comment:
Perhaps it would make more sense to use an explicit word-boundary match, so something like `s.regexpMatch(".*\\b" + wrong + "\\b.*")`.

Reply:
Good point.  
But I think I'll wait with that until we encounter FPs.  

With the current approach we also catch when the bad spelling is used as a prefix/suffix.  
(But luckily English is not like Danish, so that will be rare).